### ISSUE TITLE
`install` command

### ISSUE THREAD

**Issue by gilgamec** [swarm-game/swarm#92]
**Is your feature request related to a problem? Please describe.**

There seems to be no way to install devices on a robot once it's constructed. I'm not sure how this might be useful with regular robots, but in particular it'd be nice to be able to use eg. `if` statements or recursion at the base REPL.

**Describe the solution you'd like**

A command (say `install`) that would let a device be added to the robot's `Installed devices`.

**Describe alternatives you've considered**

For my purposes, just give more capabilities to the base by default would do it; however, I'm not ruling out the possibility that the more general option might be useful.

**Comment by byorgey** [comment]
Thanks!  I agree, this is definitely something we want.  In fact I was going to create an issue for this today but you beat me to it. =)

I am thinking along the lines of:
* The type would be `install : string -> string -> cmd ()` where the first argument is the name of the robot to install on, and the second is the name of the device to install.
* There's no restriction on what kinds of things you can install.  If you want to install a `tree`, go right ahead.  It just may not convey any capabilities.
* Installing should require some kind of device (`arc welder`, perhaps?) that is installed on the base by default.
* A potentially controversial decision is whether you should be able to `install` a device on yourself.  I am leaning towards no.  But you can certainly build another robot to install a device on you.  Why not allow self-installation? Simply because I think installing extra devices onto the `base` should be something really cool that you only get to do in the later game (e.g. you can install `treads` on the base and drive it around!).  We can accomplish this by (1) starting out the base with an installer device but (2) making installer devices difficult to construct.

**Comment by gilgamec** [comment]
That's great, but I was actually more concerned about programmatic capabilities. If `install`ing things on the base is intended to be a late-game thing, then that means that we can make robots with `cond` and `recursion` long before we can do it at the base, which seems backwards.

**Comment by byorgey** [comment]
> we can make robots with `cond` and `recursion` long before we can do it at the base, which seems backwards.

Oh, of course! You're 100% right. Let's just make it possible to install devices on yourself then. And maybe just let a grabber device do it.

**Comment by xsebek** [comment]
Just to add ideas:
- we could make `(0,0)` special square with unpickable entity `"static life support"` and if you drive away early... :skull: 
- base could require something better like tank tracks
- driving the big base around is... so... sloooow... (wait 100 and can Ctrl+C the whole computation)

**Comment by byorgey** [comment]
I created a new issue to track the idea of preventing the base from being able to move too early in the game: #334.  I'm going to close this issue since nominally it is just about adding an `install` command, which we did.
